### PR TITLE
feat: Upgrade to rollups-node alpha.12, enhance app/epoch status UI, and restrict unsupported connections

### PR DESCRIPTION
**Pull request by brunomenezes**
# Summary

This PR aligns the Explorer with the rollups-node 2.0.0-alpha.12 release and refactors how application and epoch health states are displayed in the UI.
Key outcomes include safer connection handling (blocking connections to unsupported node versions) and richer visual feedback for application and epoch lifecycles (including foreclosure indicators and diagnostic reasons).

## Why

The 2.0.0-alpha.12 update introduces critical API schema changes and new status semantics. Without these changes:
* The Explorer might mistakenly auto-connect to unsupported user-configured nodes.
* UI indicators would fail to accurately represent the updated application and epoch lifecycles.
* Storybook component tests would fall out of sync with the latest data model and visual states.

### 1. Connection & Version Hardening
* Version Bump: Raised the minimum supported node version range from ^2.0.0-alpha.9 to ^2.0.0-alpha.12 (and updated associated tests).
* Connection Filtering: The app now bypasses unsupported user-created nodes during auto-selection and skips unnecessary node info fetching for them.
* UI Restrictions: Connection cards now highlight unsupported versions in red, disable the connect button, and display explicit unsupported version text.
* Status Architecture: Refactored connection health rendering into a dedicated status-to-UI configuration mapper, adding safe fallbacks for noop (no connection) and unknown states.
### 2. Application & Epoch UI Enhancements
* Data Model Migration: Transitioned application cards from the legacy state field to the new status + enabled model.
* New Status Badges: Added dedicated visual badges for enablement (enabled/disabled), foreclosure, processed input counts, and non-OK statuses.
* Diagnostic Info: Implemented an optional "reason" section using an alert and spoiler component to cleanly handle long error/diagnostic messages.
* Foreclosure Logic: Introduced a utility to detect application foreclosure based on forecloseBlock and transaction hash validity.
* Color & State Mapping: Expanded theme colors (info and foreclosed) and updated epoch color mapping to support CLAIM_STAGED and CLAIM_FORECLOSED.
* Layout Fix: Made application address text word-break-safe to prevent layout overflow.
### 3. Testing & Storybook Coverage
* Expanded application and epoch stories to fully cover the new alpha.12 data models, including edge cases like failed/diverged/corrupted states, long diagnostic reasons, and various claim milestones (CLAIM_STAGED, CLAIM_FORECLOSED, etc.).

## Behavioral Impact
* Security & Stability: Users are blocked from connecting to unsupported nodes, and the system avoids auto-selecting them.
* UX Clarity: Application cards offer much deeper operational insight (health, foreclosure, and error reasons), and epoch states fully map to the new alpha.12 lifecycle.

## Risks & Compatibility
* Hard Dependency: Strictly coupled to rollups-node 2.0.0-alpha.12. Nodes below this range are explicitly blocked.

## Testing Summary
* Updated unit tests for version validation.
* Verified all new visual states, long text handling, and foreclosure edge cases via expanded Storybook scenarios.

### CODE REVIEW

File: apps/explorer/src/components/application/utils.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * @returns
+ */
+export const isForeclosed = (application: Application) => {
+    return (
+        application.forecloseBlock !== 0n &&
+        hasForecloseTransaction(application.forecloseTransaction)
+    );
+};
+
+const hasForecloseTransaction = (hash: Hash | null | undefined) => {
```
Comment (guidanoli) [inline apps/explorer/src/components/application/utils.ts:20]: Why does `hasForecloseTransaction(undefined)` return `true`?

File: apps/explorer/src/components/application/ApplicationCard.tsx [hunk 2]
```diff
@@ -1,35 +1,55 @@
...
 import useRightColorShade from "../../hooks/useRightColorShade";
 import { pathBuilder } from "../../routes/routePathBuilder";
 import { useSelectedNodeConnection } from "../connection/hooks";
 import SendMenu from "../send/SendMenu";
+import { isForeclosed } from "./utils";
 
 type ApplicationCardProps = { application: Application };
 
-const getStateColour = (state: ApplicationState) => {
+const getStateColour = (state: ApplicationStatus) => {
```
Comment (guidanoli) [inline apps/explorer/src/components/application/ApplicationCard.tsx]: Since `ApplicationState` was renamed as `ApplicationStatus`, shouldn't `getStateColour` be renamed as `getStatusColour`?

File: apps/explorer/src/components/application/utils.ts (hunk 1 above)
Comment (brunomenezes) [reply apps/explorer/src/components/application/utils.ts:20]: Fixed on 753b874a0488cc9f5d8b0a55ab43061c4df3b9b5

File: apps/explorer/src/components/application/ApplicationCard.tsx (hunk 2 above)
Comment (brunomenezes) [reply apps/explorer/src/components/application/ApplicationCard.tsx]: Fixed on 44be006e230c553f9acb3430b2e5bd5d1d522362

**Review by guidanoli — APPROVED** [review]
